### PR TITLE
Add cosmetic overclocks

### LINKED ISSUE #94: [Proposal] - Add cosmetic overclocks
it could be in the same UI element as the weapon OCs.
currently, that element looks like this:
```
.
├── Dwarf1
│   ├── Weapon1
│   │   ├── Overclock1
│   │   ├── Overclock2
│   │   └── Overclock3
│   ├── Weapon2
│   └── Weapon3
├── Dwarf2
├── Dwarf3
└── Dwarf4
```

and after implementing:

```
.
├── Weapon
│   ├── Dwarf1
│   │   ├── Weapon1
│   │   │   ├── Overclock1
│   │   │   ├── Overclock2
│   │   │   └── Overclock3
│   │   ├── Weapon2
│   │   └── Weapon3
│   ├── Dwarf2
│   ├── Dwarf3
│   └── Dwarf4
└── Cosmetic
    ├── Overclock1
    │   ├── Dwarf1
    │   ├── Dwarf2
    │   ├── Dwarf3
    │   └── Dwarf4
    ├── Overclock2
    └── Overclock3
```
bearing in mind that each cosmetic is unlocked separately for each dwarf.
https://deeprockgalactic.wiki.gg/wiki/Accessory_Shop#Matrix_Core_Cosmetics
https://deeprockgalactic.wiki.gg/wiki/Weapon_Skins

### PR DESCRIPTION
**Pull request by wahlp**
closes #94

### CODE REVIEW

**Comment by wahlp** [comment]
adding/removing weapon/cosmetic OCs works now.

i'm considering breaking up the UI element for the Cosmetic category further, it has 87 items which is a lot

File: src/main/python/helpers/datatypes.py [hunk 1]
```diff
@@ -27,9 +28,9 @@ def __add__(self, other):
 class Item:
     """An item read from the known list of GUIDs"""
 
-    type_: Literal["weapon", "cosmetic"]
-    dwarf: str
+    category: str
```
Comment (AnthonyMichaelTDM) [inline src/main/python/helpers/datatypes.py:31]: can category be a StrEnum instead of a string? (same with dwarf and status really)

Comment (wahlp) [reply src/main/python/helpers/datatypes.py:31]: sure, i'll look into it

**Comment by wahlp** [comment]
the filter is broken only on the weapon tree element, not sure why

**Comment by wahlp** [comment]
@AnthonyMichaelTDM it's ready for review

**Comment by simon-wg** [comment]
Upon trying it out it seems state_manager.get_unacquired_overclocks() returns an empty list on my save with all weapon overclocks unlocked.

**Comment by simon-wg** [comment]
I think I've found the issue, pushing a solution soon

**Review by AnthonyMichaelTDM — APPROVED** [review]
lgtm

**Pull request merged**